modules/core/src/main/scala/com.snowplowanalytics/iglu.schemaddl/bigquery/Suggestion.scala

```diff
@@ -40,19 +41,55 @@ object Suggestion {
   val integerSuggestion: Suggestion = (schema, required) =>
     schema.`type` match {
       case Some(CommonProperties.Type.Integer) =>
-        Some(name => Field(name, Type.Integer, Mode.required(required)))
+        Some(name => Field(name, Type.fromGenericType(decimals.integerType(schema)), Mode.required(required)))
       case Some(CommonProperties.Type.Union(types)) if withNull(types, CommonProperties.Type.Integer) =>
-        Some(name => Field(name, Type.Integer, Mode.Nullable))
+        Some(name => Field(name, Type.fromGenericType(decimals.integerType(schema)), Mode.Nullable))
       case _ => None
     }
 
+  val numericSuggestion: Suggestion = (schema, required) => schema.`type` match {
+    case Some(CommonProperties.Type.Number) =>
+      schema.multipleOf match {
+        case Some(NumberProperty.MultipleOf.IntegerMultipleOf(_)) =>
+          Some(name => Field(name, Type.fromGenericType(decimals.integerType(schema)), Mode.required(required)))
+        case Some(mult: NumberProperty.MultipleOf.NumberMultipleOf) =>
+          Some(name => Field(name, numericWithMultiple(mult, schema.maximum, schema.minimum), Mode.required(required)))
+        case None => None
+      }
+    case Some(CommonProperties.Type.Union(types)) if withNull(types, CommonProperties.Type.Number) =>
+      schema.multipleOf match {
+        case Some(NumberProperty.MultipleOf.IntegerMultipleOf(_)) =>
+          Some(name => Field(name, Type.fromGenericType(decimals.integerType(schema)), Mode.Nullable))
+        case Some(mult: NumberProperty.MultipleOf.NumberMultipleOf) =>
+          Some(name => Field(name, numericWithMultiple(mult, schema.maximum, schema.minimum), Mode.Nullable))
+        case None => None
+      }
```

Comment:
Could you remove this entire block starting from line 59, and let everything fall through to the next `case` on line 67?

Reply:
It won't fall though there, check the onlyNumeric implementation L180.

Reply:
Ah, I see.  OK how about this then:

```scala
case Some(CommonProperties.Type.Union(types)) if withNull(types, CommonProperties.Type.Number) || onlyNumeric(types, true) =>
```